### PR TITLE
[CPU] Pad node migration on nGraph

### PR DESCRIPTION
**Pull request by mandrono**

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp [hunk 1]
```diff
@@ -13,54 +12,90 @@
...
 using namespace MKLDNNPlugin;
 using namespace InferenceEngine;
 
-MKLDNNPadNode::MKLDNNPadNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
-        : MKLDNNNode(layer, eng, cache) {}
+bool MKLDNNPadNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto pad = std::dynamic_pointer_cast<const ngraph::opset1::Pad>(op);
+        if (!pad) {
+            errorMessage = "CPU plug-in supports Pad node only from opset1. Node name: " + op->get_friendly_name();
```
[the remark this answers is not in the document]
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp]: fixed

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp [hunk 2]
```diff
@@ -13,54 +12,90 @@
...
+                padValue = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(pad->get_input_node_shared_ptr(PAD_VALUE_ID))->cast_vector<float>()[0];
+            }
+        } else if (pad_mode == ngraph::op::PadMode::EDGE) {
+            padMode = EDGE;
+        } else if (pad_mode == ngraph::op::PadMode::REFLECT) {
+            padMode = REFLECT;
+        } else if (pad_mode == ngraph::op::PadMode::SYMMETRIC) {
+            padMode = SYMMETRIC;
+        } else {
+            THROW_IE_EXCEPTION << errorPrefix << " has incorrect pad_mode. Only constants|edge|reflect|symmetric modes are supported!";
```
[the remark this answers is not in the document]
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp]: fixed

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp [hunk 3]
```diff
@@ -13,54 +12,90 @@
...
-    if (padLayer == nullptr)
-        THROW_IE_EXCEPTION << "Cannot convert Pad layer.";
+MKLDNNPadNode::MKLDNNPadNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
+        : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    if (isSupportedOperation(op, errorMessage)) {
+        errorPrefix = "Pad node with name '" + op->get_friendly_name() + "'";
+        const auto pad = std::dynamic_pointer_cast<const ngraph::opset1::Pad>(op);
+
+        const auto pb = pad->get_pads_begin();
```
[the remark this answers is not in the document]
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp:54]: fixed

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
Please resolve conflicts

**Pull request merged**